docs/intents/getQuote.md

```diff
@@ -0,0 +1,363 @@
+
+Our approach focuses on creating a common language for cross-chain quote requests and responses, allowing wallets and applications to work seamlessly with any compliant bridge protocol.
+
+## **Purpose**
+
+The primary purpose of this standardization effort is to **eliminate the integration complexity** that currently exists in the cross-chain ecosystem. By establishing a common interface for quote endpoints, we aim to:
+
+- **Reduce Development Overhead**: Wallets and dApps can integrate with multiple bridge protocols using a single, consistent API interface
+- **Enable Seamless Bridge Comparison**: Allow applications to easily compare quotes from different providers without maintaining separate adapters
+- **Improve User Experience**: Provide consistent and predictable quote information across all bridge protocols
+- **Foster Ecosystem Interoperability**: Create a foundation for cross-bridge collaboration and competition based on performance rather than integration difficulty
+
+# In-Depth
+
+## **Core Design Decision: Server-Side vs Client-Side Route Calculation**
+
+A fundamental architectural decision for the standardized `getQuote` endpoint is determining **where the route optimization occurs**. We have identified two primary approaches, each with distinct trade-offs:
+
+## **Approach 1: Server-Side Route Calculation**
+
+In this approach, the endpoint receives user balance information and returns **a single optimized route** that the wallet should execute. The server can implement various routing algorithms including **Greedy Algorithm**, **Dynamic Programming**, **Dijkstra's Algorithm**, **Multi-Objective Optimization**, **Genetic Algorithm**, **Simulated Annealing**, **Ant Colony Optimization**, and *A Search Algorithm**.
+
+### **Input Schema:**
+
+```tsx
+interface ServerSideQuoteRequest {
+  // Map of chain ID to user's token balance on that chain
+  // chainId: numeric blockchain identifier (e.g., 1 = Ethereum, 137 = Polygon)
+  // balance: amount as integer string in wei (smallest unit of the token)
+  // tokenAddress: token contract address (hex string, e.g., "0x...")
+  // priority: which token the user prefeer to spend (e.g., 0 = Max Priority)
+  userBalances: Map<chainId, {balance, tokenAddress, priority}>
+
+  // Desired amount to receive on target chain (as integer string in wei)
+  targetAmount: string
+
+  // Destination blockchain ID where user wants to receive tokens
+  targetChain: chainId
+
+  // Destination token contract address (hex string, e.g., "0x...")
+  targetTokenAddress: string
+
+  // Optional: specify which routing algorithm to use
+  algorithm?: 'greedy' | 'dynamic-programming' | 'dijkstra' | 'multi-objective' | 'genetic' | 'simulated-annealing' | 'ant-colony' | 'a-star'
```

Comment:
If the server does not support this configuration, should it return an error or simply ignore it?

Reply:
We could return a NotValidAttribute error, and if the algorithm attribute is valid but the algorithm is not there, we could return a NotFoundAlgorithm error, mentioning the implemented algorithms